### DISCUSSION TITLE
possible to insert recommendation text?

### DISCUSSION THREAD

**Discussion by Coolroo** [microsoft/vscode-discussions#530] [Extension Development QnA] [answered]
Hi there! Does anyone know if it's possible to insert multiple blocks of text that are not really there. What I mean by this is it's essentially an InlineCompletionItem, but it can be anywhere in the document, and isn't actually saved to the disk when the file is saved.

**Comment by r3m0t** [comment, marked as answer]
Sounds like a text decoration with `after.contentText` property set.

**Reply by Coolroo** [reply]
> Sounds like a text decoration with `after.contentText` property set. 

I tried this but i need to display multi line text, and it doesn't seem to work with the decorator, unless I span it over areas that already have text. What I'm hoping for is something like copilot where it displays the text and adjusts the text around it, but doesn't actually insert it
